### PR TITLE
RXI-471 switch for batch / single attestations mode

### PR DESCRIPTION
**Pull request by oleks-rip**

### CODE REVIEW

File: src/xbwd/federator/Federator.h [hunk 1]
```diff
@@ -64,12 +67,131 @@ struct Submission
...
+struct SubmissionBatch : public Submission
+{
+    ripple::STXChainAttestationBatch batch_;
+
+    SubmissionBatch(
         uint32_t lastLedgerSeq,
         uint32_t accountSqn,
         ripple::STXChainAttestationBatch const& batch);
+
+    virtual std::pair<std::string, std::string>
```
Comment (seelabs) [inline src/xbwd/federator/Federator.h:119]: It's not obvious what the return type of this function means. Can you add a comment?

**Review by seelabs — COMMENTED** [review]
Left some minor comments.

File: src/xbwd/federator/Federator.h [hunk 2]
```diff
@@ -164,6 +286,8 @@ class Federator : public std::enable_shared_from_this<Federator>
     ChainArray<std::deque<FederatorEvent>> replays_;
     beast::Journal j_;
 
+    config::Config config_;
```
[the remark this answers is not in the document]
Comment (oleks-rip) [reply src/xbwd/federator/Federator.h]: Done 5f4feba4e1cc

File: src/xbwd/federator/Federator.h (hunk 1 above)
Comment (oleks-rip) [reply src/xbwd/federator/Federator.h:122]: Done 5f4feba4e1cc

File: src/xbwd/rpc/RPCHandler.cpp [hunk 3]
```diff
@@ -149,10 +149,25 @@ doSelectAll(
...
+        auto const& config(app.config());
+        if (config.useBatch)
+        {
+            ripple::STXChainAttestationBatch batch{
+                bridge, claims.begin(), claims.end()};
+            result[ripple::sfXChainAttestationBatch.getJsonName()] =
+                batch.getJson(ripple::JsonOptions::none);
+        }
+        else
+#endif
```
[the remark this answers is not in the document]
Comment (oleks-rip) [reply src/xbwd/rpc/RPCHandler.cpp]: Done 5f4feba4e1cc

**Comment by seelabs** [comment]
merged

**Pull request closed without merging**